core/common/src/main/java/alluxio/security/authentication/DefaultAuthenticationServer.java

```diff
@@ -0,0 +1,226 @@
+package alluxio.security.authentication;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.exception.status.UnauthenticatedException;
+import alluxio.grpc.SaslAuthenticationServiceGrpc;
+import alluxio.grpc.SaslMessage;
+import alluxio.resource.LockResource;
+import alluxio.util.SecurityUtils;
+import io.grpc.ServerInterceptor;
+import io.grpc.stub.StreamObserver;
+import net.jcip.annotations.GuardedBy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.security.sasl.SaslException;
+import javax.security.sasl.SaslServer;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Default implementation of {@link AuthenticationServer}. Its functions include: -> Authentication
+ * server against which client channels could get authenticated -> Registry for identity for known
+ * channels during RPC calls.
+ *
+ */
+public class DefaultAuthenticationServer
+    extends SaslAuthenticationServiceGrpc.SaslAuthenticationServiceImplBase
+    implements AuthenticationServer {
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultAuthenticationServer.class);
+
+  @GuardedBy("mClientsLock")
+  /** List of channels authenticated against this server. */
+  protected final Map<UUID, AuthenticatedChannelInfo> mChannels;
```

Comment:
Instead of guarding this with a lock, would a concurrent map provide us with sufficient guarantees?